src/operation_dispatcher.rs

```diff
@@ -171,25 +168,18 @@ impl OperationDispatcher {
         }
     }
 
-    pub fn build_operations(&mut self, actions: &[Action]) -> Result<(), OperationError> {
+    pub fn build_operations(&mut self, actions: &[Rc<RuntimeAction>]) {
         let mut operations: Vec<Rc<Operation>> = vec![];
         for action in actions.iter() {
-            if let Some(service) = self.service_handlers.get(&action.service) {
-                operations.push(Rc::new(Operation::new(
-                    service.get_service(),
-                    action.clone(),
-                    Rc::clone(service),
-                )))
-            } else {
-                error!("Unknown service: {}", action.service);
-                return Err(OperationError::new(
-                    Status::ParseFailure,
-                    Default::default(),
-                ));
-            }
+            operations.push(Rc::new(Operation::new(
+                Rc::clone(action),
+                GrpcServiceHandler::new(
+                    Rc::clone(&action.grpc_service()),
+                    Rc::clone(&self.header_resolver),
+                ),
+            )));
```

Comment:
> the original implementation passed references to GrpcServiceHandlers and only created as many as we had services, instead of one per operation as we have now

Correct. The reason being that the runtime action now has the grpc_service embedded, as part of it. It could even be the GrpcServiceHandler itself, why not? It happens that creating GrpcServiceHandler is cheap as it is composed of a couple of `Rc<T>` fields. The benefit is to remove the `service_handlers: HashMap<String, Rc<GrpcServiceHandler>>` from the `OperationDispatcher` and remove the need to check at request time if the service exist or not. Each action, which are built at config time, has the assigned service and grpc_service instance ready to be used. For example, if there are 10 services defined in the config, but actions only use 2 of them why keep them  all in `OperationDispatcher` ? 

That is the reasoning. Let me know if that makes sense or not.

Reply:
Yeah, something is still fishy with these `Rc`s tbh... if anthing tho, this `Rc::clone` is unnecessary, the callee is already doing that, and returning an `Rc` for the caller to own. That just literally just increments the counter by one, to drop the returned reference back on the floor and as such decrementing the counter. That being said the compiler _may_ opt this out... yet, it leave the unnecessary noise for the reader to parse out.

Reply:
good catch _alexclippy_

Fixed the unnecessary RC clone